framework/bundles/org.jboss.tools.rsp.server.spi/src/main/java/org/jboss/tools/rsp/server/spi/client/ClientThreadLocal.java

```diff
@@ -0,0 +1,35 @@
+package org.jboss.tools.rsp.server.spi.client;
+
+import org.jboss.tools.rsp.api.RSPClient;
+
+
+/**
+ * This class is to keep track of which client is actively 
+ * making a request on the server. 
+ * 
+ * While many tutorials online indicate not to use thread-local
+ * with an executor service, and while the lsp launchers do use
+ * an executor service, the fact is that each thread reads one
+ * socket until that socket is closed, at which point the thread
+ * may be repurposed for a new connection.
+ * 
+ * With that in mind, there can only possibly be one client per 
+ * thread at any given time. 
+ * 
+ * Clients can use this to discover which client is making a given request
+ * or to ensure all clients are authenticated for various purposes.
+ * 
+ * @author rob
+ *
+ */
+public class ClientThreadLocal {
+	 private static ThreadLocal<RSPClient> activeClient = new ThreadLocal<>();
```

Comment:
we should add a private constructor to this class preventing user code from instantiating it